SDKBuildScripts/Update-PlayFabSdk.ps1

```diff
@@ -0,0 +1,160 @@
+[CmdletBinding(SupportsShouldProcess = $true, DefaultParameterSetName = "ApiSpecGitUrl")]
+param(
+    [Parameter(Position = 1)]
+    [ValidateSet(
+        "ActionScriptSDK",
+        "Cocos2d-xSDK",
+        "CSharpSDK",
+        "JavaSDK",
+        "JavaScriptSDK",
+        "LuaSDK",
+        "NodeSDK",
+        "Objective_C_SDK",
+        "PhpSDK",
+        "PostmanCollection",
+        "PythonSDK",
+        "SdkTestingCloudScript",
+        "UnrealMarketplacePlugin",
+        "UnitySDK",
+        "WindowsSDK",
+        "XPlatCppSDK")]
+    [string[]]$SdkNames,
+    [Parameter(ParameterSetName="ApiSpecPath")]
+    [string]$ApiSpecPath,
+    [Parameter(ParameterSetName="ApiSpecPfUrl")]
+    [string]$ApiSpecPfUrl,
+    [Parameter(ParameterSetName="ApiSpecPfUrlBranch")]
+    [string]$ApiSpecBranch,
+    [Parameter(ParameterSetName="ApiSpecGitUrl")]
+    [switch]$ApiSpecGitUrl,
+    [switch]$KeepSource,
+    [switch]$NonNullable,
+    [switch]$Beta
+)
+
+begin
+{
+    if(!(Get-Command node))
+    {
+        throw "You must have Node.js installed to generate the SDK"
+    }
+
+    $sdkTargetSrcMap = @{
+        "ActionScriptSDK" = "actionscript";
+        "Cocos2d-xSDK" = "cpp-cocos2dx";
+        "CSharpSDK" = "csharp";
+        "JavaSDK" = "java";
+        "JavaScriptSDK" = "javascript";
+        "LuaSDK" = "LuaSdk";
+        "NodeSDK" = "js-node";
+        "Objective_C_SDK" = "objc";
+        "PhpSDK" = "PhpSdk";
+        "PostmanCollection" = "postman";
+        "PythonSDK" = "PythonSdk";
+        "SdkTestingCloudScript" = "SdkTestingCloudScript";
+        "UnrealMarketplacePlugin" = "UnrealMarketplacePlugin";
+        "UnitySDK" = "unity-v2";
+        "WindowsSDK" = "windowssdk";
+        "XPlatCppSDK" = "xplatcppsdk";
+    }
+
+    $sdksPath = Resolve-Path (Join-Path $PSScriptRoot "..\..\sdks")
+    if(!(Test-Path $sdksPath))
+    {
+        mkdir $sdksPath
+    }
+
+    if($ApiSpecBranch)
+    {
+        $ApiSpecPfUrl = "https://$($ApiSpecBranch).playfabapi.com/apispec"
+    }
+
+    if($ApiSpecPath)
+    {
+        if(!(Test-Path $ApiSpecPath))
+        {
+            throw "Unable to find ApiSpecPath '$ApiSpecPath'."
+        }
+
+        $apiSpecSource = "-apiSpecPath $ApiSpecPath"
+    }
+    elseif($ApiSpecPfUrl)
+    {
+        if(![Uri]::IsWellFormedUriString($ApiSpecPfUrl, "Absolute"))
+        {
+            throw "You must specify a valid URL for ApiSpecPfUrl.  Unable to parse '$ApiSpecPfUrl' as a URL."
+        }
+
+        $apiSpecSource = "-apiSpecPfUrl $ApiSpecPfUrl"
+    }
+    else
+    {
+        $apiSpecSource = "-apiSpecGitUrl"
+    }
+}
+
+process
+{
+    foreach($sdkName in $SdkNames)
+    {
+        $targetSrc = $sdkTargetSrcMap[$sdkName]
```

Comment:
For non-public repos, it's important to be able to accept this information as a parameter, if it's not present in the public dictionary.

Reply:
I don't quite understand how this is used.  Will the user still need to pass in `$sdkName` CSharpSDK and then some other local path?  Or does it make sense to just use sdkName as targetSrc if it doesn't exist in the map?

This is passed into generate.js as:
```
node generate.js "$targetSrc=$destPath" ...
```

What does this represent when the user provides a value?

Reply:
Replied privately.